apps/taxes/src/pages/api/webhooks/order-created.ts

```diff
@@ -0,0 +1,108 @@
+import { SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+import {
+  OrderCreatedEventSubscriptionFragment,
+  OrderStatus,
+  UntypedOrderCreatedSubscriptionDocument,
+  UpdateMetadataDocument,
+  UpdateMetadataMutation,
+  UpdateMetadataMutationVariables,
+} from "../../../../generated/graphql";
+import { saleorApp } from "../../../../saleor-app";
+import { createLogger } from "../../../lib/logger";
+import { getActiveTaxProvider } from "../../../modules/taxes/active-tax-provider";
+import { createClient } from "../../../lib/graphql";
+import { Client } from "urql";
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+type OrderCreatedPayload = Extract<
+  OrderCreatedEventSubscriptionFragment,
+  { __typename: "OrderCreated" }
+>;
+
+export const orderCreatedAsyncWebhook = new SaleorAsyncWebhook<OrderCreatedPayload>({
+  name: "OrderCreated",
+  apl: saleorApp.apl,
+  event: "ORDER_CREATED",
+  query: UntypedOrderCreatedSubscriptionDocument,
+  webhookPath: "/api/webhooks/order-created",
+});
+
+// * This is the key that we use to store the provider order id in the Saleor order metadata.
+export const PROVIDER_ORDER_ID_KEY = "externalId";
+
+// * We need to store the provider order id in the Saleor order metadata so that we can
+// * update the provider order when the Saleor order is fulfilled.
+async function updateOrderMetadataWithExternalId(
+  client: Client,
+  orderId: string,
+  externalId: string
+) {
+  const { error } = await client
+    .mutation<UpdateMetadataMutation>(UpdateMetadataDocument, {
+      id: orderId,
+      input: [
+        {
+          key: PROVIDER_ORDER_ID_KEY,
+          value: externalId,
+        },
+      ],
+    } as UpdateMetadataMutationVariables)
+    .toPromise();
```

Comment:
why not metadata manager?

Reply:
This is order metadata, not app

Reply:
Okay, didnt see the function name. However, graphql mutation name can be also more explicit, it can also be conflicted easily